api/internal/target/kusttarget.go

```diff
@@ -69,6 +70,15 @@ func (kt *KustTarget) Load() error {
 	if err != nil {
 		return err
 	}
+
+	// show warning message when using deprecated fields.
+	warningMessages := k.CheckDeprecatedFields()
```

Comment:
Are we sure this is going to work, given that `FixKustomizationPreUnmarshalling` above mutates the file to accommodate some of the deprecated fields?

Reply:
Looks like `FixKustomizationPreUnmarshalling` is fixing for raw `[]byte` data before unmarshaling to yaml.
https://github.com/kubernetes-sigs/kustomize/blob/f61b075d3bd670b7bcd5d58ce13e88a6f25977f2/api/types/fix.go#L12-L32

I think this function doesn't affect some of the deprecated fields in `Kustomization` struct.

Reply:
Interestingly, this changes `imageTags` to `images`, which is a deprecation I wasn't even aware of. Also, it makes it seem like the field called `patchesStrategicMerge` was originally called `patches`, and we later re-introduced `patches` with a different meaning. 🤦 

Ideally, I think we should warn about `imageTags` too, so we can get rid of `FixKustomizationPreUnmarshalling` along with all the deprecated fields.

Also, I wonder if folks using the super old version of `patches` would get a weird warning about `patchesStrategicMerge` (which they aren't using) because of this method. Can we check this please?

Reply:
> Ideally, I think we should warn about imageTags too, so we can get rid of FixKustomizationPreUnmarshalling along with all the deprecated fields.

I think it is hard to delete the `FixKustomizationPreUnmarshalling` function when we support the super old version of patches.
Because super old `patches` (It is called `patchesStrategicMerge` today) have a different struct to today's `patches` field.
I think we will get an error when unmarshaling `[]byte` to `Kustomization` struct if we delete pre unmarshalling process.

https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/kustomization.go#L66-L76
https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/patchstrategicmerge.go#L9
https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/patch.go#L8-L24

Reply:
> Ideally, I think we should warn about imageTags too, so we can get rid of FixKustomizationPreUnmarshalling along with all the deprecated fields.

I think it is hard to delete the `FixKustomizationPreUnmarshalling` function when we support the super old version of patches.
Because super old `patches` (It is called `patchesStrategicMerge` today) have a different struct to today's `patches` field.
I think we will get an error when unmarshaling `[]byte` to `Kustomization` struct if we delete pre unmarshalling process.

https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/kustomization.go#L66-L76
https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/patchstrategicmerge.go#L9
https://github.com/kubernetes-sigs/kustomize/blob/4fc02497ae7de4b1f7e31656e13720d4e719dcc0/api/types/patch.go#L8-L24

Reply:
> Also, I wonder if folks using the super old version of patches would get a weird warning about patchesStrategicMerge (which they aren't using) because of this method. Can we check this please?

That is right. If we use very old style `patches`,  the warning message of `patchesStrategicMerge` appear.